### PR TITLE
CB-18196: use the new entitlement for the long time backup process

### PR DESCRIPTION
**Pull request by RokolyA94**
CB-18196: use the new entitlement for the long time backup process

### CODE REVIEW

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/model/Entitlement.java [hunk 1]
```diff
@@ -108,5 +108,6 @@ public enum Entitlement {
     CDP_ROTATE_SALTUSER_PASSWORD,
     CDP_USERSYNC_ENFORCE_GROUP_MEMBER_LIMIT,
     CDP_DATA_LAKE_BACKUP_RESTORE_PERMISSION_CHECKS,
-    CDP_DATAHUB_EXPERIMENTAL_SCALE_LIMITS
+    CDP_DATAHUB_EXPERIMENTAL_SCALE_LIMITS,
+    CDP_DATALAKE_BACKUP_ON_RESIZE_LONG_TIMEOUT
```
Comment (kkalvagadda1) [inline auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/model/Entitlement.java:112]: This entitlement is to support backups that take longer. This is not related to resizing.

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java [hunk 2]
```diff
@@ -56,19 +64,25 @@ protected Selectable doAccept(HandlerEvent<DatalakeFullBackupWaitRequest> event)
...
             LOGGER.info("Start polling datalake full backup status for id: {}", sdxId);
-            PollingConfig pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, durationInMinutes,
-                TimeUnit.MINUTES);
+            if (entitlementService.isLongTimeBackupEnabled(ThreadBasedUserCrnProvider.getAccountId())) {
+                pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, longDurationInMinutes,
+                        TimeUnit.MINUTES);
+            } else {
+                pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, durationInMinutes,
+                        TimeUnit.MINUTES);
+            }
```
Comment (kkalvagadda1) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java:75]: How about simplifying the change by adding below? The rest of the code need not change.
```
int durationInMinutes 
 = (entitlementService.isLongTimeBackupEnabled(ThreadBasedUserCrnProvider.getAccountId()) ? longDurationInMinutes : durationInMinutes;
```

**Review by kkalvagadda1 — CHANGES_REQUESTED** [review]

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/model/Entitlement.java (hunk 1 above)
Comment (RokolyA94) [reply auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/model/Entitlement.java:112]: fixed

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java (hunk 2 above)
Comment (RokolyA94) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java]: modified

Comment (RokolyA94) [reply datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java]: modified

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java [hunk 3]
```diff
@@ -36,9 +38,15 @@ public class DatalakeFullBackupWaitHandler extends ExceptionCatcherEventHandler<
     @Value("${sdx.stack.backup.status.duration_min:240}")
     private int durationInMinutes;
 
+    @Value("${sdx.stack.backup.status.long_darution_min:840}")
```
Comment (sxxgrc) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java:41]: Typo: `duration`

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java [hunk 4]
```diff
@@ -56,19 +64,21 @@ protected Selectable doAccept(HandlerEvent<DatalakeFullBackupWaitRequest> event)
...
+                    pollingConfig, "Full backup");
             response = new DatalakeBackupSuccessEvent(sdxId, userId, request.getOperationId());
         } catch (UserBreakException userBreakException) {
             LOGGER.info("Full backup polling exited before timeout. Cause: ", userBreakException);
             response = new DatalakeBackupFailedEvent(sdxId, userId, userBreakException);
         } catch (PollerStoppedException pollerStoppedException) {
             LOGGER.info("Full backup poller stopped for cluster: {}", sdxId);
             response = new DatalakeBackupFailedEvent(sdxId, userId,
-                new PollerStoppedException("Datalake backup timed out after " + durationInMinutes + " minutes"));
+                    new PollerStoppedException("Datalake backup timed out after " + durationInMinutes + " minutes"));
```
Comment (sxxgrc) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java:81]: Should this not be `duration` now instead of `durationInMinutes`?

**Review by kkalvagadda1 — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java [hunk 5]
```diff
@@ -55,20 +63,23 @@ protected Selectable doAccept(HandlerEvent<DatalakeFullBackupWaitRequest> event)
         Long sdxId = request.getResourceId();
         String userId = request.getUserId();
         Selectable response;
+        int duration = 0;
         try {
+            duration = entitlementService.isLongTimeBackupEnabled(ThreadBasedUserCrnProvider.getAccountId())
```
Comment (lacikaaa) [inline datalake/src/main/java/com/sequenceiq/datalake/flow/dr/backup/handler/DatalakeFullBackupWaitHandler.java:68]: could you move this to the variable declaration?

**Review by lacikaaa — APPROVED** [review]

**Review by lacikaaa — CHANGES_REQUESTED** [review]
please ensure you have only one commmit

**Review by lnardai — APPROVED** [review]

**Review by lacikaaa — APPROVED** [review]
